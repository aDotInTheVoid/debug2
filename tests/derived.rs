use debug2::pprint;
use insta::assert_snapshot;

macro_rules! check {
    ($e:expr) => {
        assert_snapshot!(pprint($e))
    };
}

#[derive(debug2::Debug)]
struct X {
    a: i32,
    b: i32,
}

#[derive(debug2::Debug)]
struct Wrapper<T>(T);

#[test]
fn simple() {
    let x = X { a: 1, b: 5 };

    check!(x);
}
