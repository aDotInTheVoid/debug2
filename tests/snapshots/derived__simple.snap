---
source: tests/derived.rs
expression: pprint(x)

---
X { a: 1, b: 5 }
